lib/symboldatabase.cpp

```diff
@@ -4350,6 +4350,18 @@ const Function *Function::getOverriddenFunction(bool *foundAllBaseClasses) const
     return getOverriddenFunctionRecursive(nestedIn->definedType, foundAllBaseClasses);
 }
 
+// prevent recursion if base is the same except for different template parameters
+static bool isDerivedFromItself(const std::string& thisName, const std::string& baseName)
+{
+    const auto pos = thisName.find('<');
+    if (pos == std::string::npos)
+        return false;
+    const auto posBase = baseName.find('<');
```

Comment:
I don't have a strong opinion.. but we don't really need posBase how about:

    return pos < baseName.size() && thisName.compare(0, pos + 1, baseName) == 0;

I am not sure if the `posBase < baseName.size() && ` is needed though.